### PR TITLE
fix locking issue in enter/leave entity scripts

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by birarda** [comment]
Will this still safely allow edits in those callbacks from entity scripts (since the lock is gone)?

**Comment by ZappoMan** [comment]
Yes. The scripting interface already does proper locking. The outer locks were placed incorrectly and were redundant.

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -294,6 +294,11 @@ void EntityTreeRenderer::checkEnterLeaveEntities() {
                     entitiesContainingAvatar << entity->getEntityItemID();
                 }
             }
+            _tree->unlock();
+            
+            // Note: at this point we don't need to worry about the tree being locked, because we only deal with
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: @birarda - note - this comment explains why it's safe to only lock the tree during the search and have it unlocked during the actual script event callbacks.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e7ead2ead37b13dec7db60ae5aea42894cf1370 available at: https://gist.github.com/921539997a64d84fe4b2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4515.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4515.dmg

**Pull request merged**